pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -363,17 +386,35 @@ func (a *FlavorAssigner) assignFlavors(log logr.Logger, counts []int32) Assignme
 			}
 			psAssignment.append(flavors, status)
 		}
-		if features.Enabled(features.TopologyAwareScheduling) {
-			if a.wl.Obj.Spec.PodSets[i].TopologyRequest != nil {
-				assignTopology(log, &psAssignment, a.cq, a.wl.TotalRequests[i], &a.wl.Obj.Spec.PodSets[i])
-			}
-		}
 
 		assignment.append(podSet.Requests, &psAssignment)
 		if psAssignment.Status.IsError() || (len(podSet.Requests) > 0 && len(psAssignment.Flavors) == 0) {
 			return assignment
 		}
 	}
+	if assignment.RepresentativeMode() == NoFit {
+		return assignment
+	}
+
+	if features.Enabled(features.TopologyAwareScheduling) && assignment.RepresentativeMode() == Fit {
+		tasRequests := assignment.WorkloadsTopologyRequests(a.wl, requests, a.cq)
+		result := a.cq.FindTopologyAssignmentsForWorkload(tasRequests)
+		if failure := result.Failure(); failure != nil {
+			// There is at least one PodSet which does not fit
+			psAssignment := assignment.podSetAssignmentByName(string(failure.PodSet))
+			psAssignment.reason(failure.Reason)
+			// clear flavors to make the PodSet assignment as NoFit
+			psAssignment.Flavors = nil
+			// clear the representative mode as
+			assignment.representativeMode = nil
```

Comment:
```suggestion
			assignment.representativeMode = noFit
```

Is there any reason why we do not obviously set `noFit`?

Reply:
NoFit works equally well. I just wanted to assign nil to make sure it recomputes as NoFit, but since the result is guaranteed assigning directly is faster.

Reply:
My point is different behavior for `noFit` situation in flavorassigner. In other places, assigned says obviously `noFit` 

- https://github.com/kubernetes-sigs/kueue/blob/70517e20ab5bc3b19125e535d588f7e91eed1203/pkg/scheduler/flavorassigner/flavorassigner.go#L650-L655
- https://github.com/kubernetes-sigs/kueue/blob/70517e20ab5bc3b19125e535d588f7e91eed1203/pkg/scheduler/flavorassigner/flavorassigner.go#L470

So, I would offer to keep consistency behavior for the NoFit situation.

Reply:
If we set to nil it would be recomputed to NoFit anyway. This is because representativeMode is just a cache.